dashboard/modules/job/tests/test_backwards_compatibility.py

```diff
@@ -76,19 +76,20 @@ def test_error_message():
     )
     wait_for_condition(lambda: client.get_job_status(job_id) == JobStatus.SUCCEEDED)
 
-    # `entrypoint_num_cpus`, `entrypoint_num_gpus`, and `entrypoint_resources`
-    # are not supported in ray<2.2.0.
+    # `entrypoint_num_cpus`, `entrypoint_num_gpus`, `entrypoint_memory`, and
+    # `entrypoint_resources` are not supported in ray<2.2.0.
     for unsupported_submit_kwargs in [
         {"entrypoint_num_cpus": 1},
         {"entrypoint_num_gpus": 1},
+        {"entrypoint_memory": 4},
         {"entrypoint_resources": {"custom": 1}},
     ]:
         with pytest.raises(
             Exception,
             match="Ray version 2.0.1 is running on the cluster. "
-            "`entrypoint_num_cpus`, `entrypoint_num_gpus`, and "
-            "`entrypoint_resources` kwargs"
-            " are not supported on the Ray cluster. Please ensure the cluster is "
+            "`entrypoint_num_cpus`, `entrypoint_num_gpus`, "
+            "`entrypoint_memory`, and `entrypoint_resources` kwargs "
+            "are not supported on the Ray cluster. Please ensure the cluster is "
             "running Ray 2.2 or higher.",
```

Comment:
Hmm, I think we need to update the backwards compatibility logic and test. Basically, `entrypoint_memory` will only appear in Ray 2.8, which is the next Ray release, so we need a Ray version check for that and we likely need a different error message.  Let me know if you have trouble finding or modifying the existing code for this.